### PR TITLE
Prefer default constructor implementations

### PR DESCRIPTION
**Pull request by hjmjohnson**

### CODE REVIEW

**Comment by hjmjohnson** [comment]
Circle CI failure seems unrealated:
CMake Error at Libs/vtkITK/CMakeLists.txt:64 (find_package):
  By not providing "FindvtkAddon.cmake" in CMAKE_MODULE_PATH this project has
  asked CMake to find a package configuration file provided by "vtkAddon",
  but CMake did not find one.

File: Modules/Loadable/VolumeRendering/MRMLDM/vtkMRMLVolumeRenderingDisplayableManager.cxx [hunk 1]
```diff
@@ -1079,7 +1079,7 @@ void vtkMRMLVolumeRenderingDisplayableManager::vtkInternal::FindPickedDisplayNod
 
 //---------------------------------------------------------------------------
 vtkMRMLVolumeRenderingDisplayableManager::vtkMRMLVolumeRenderingDisplayableManager()
-: VolumeRenderingLogic(nullptr)
+
```
Comment (jcfr) [inline Modules/Loadable/VolumeRendering/MRMLDM/vtkMRMLVolumeRenderingDisplayableManager.cxx:1082]: Would it be possible to remove the additional empty line like this one ?

**Comment by jcfr** [comment]
> Circle CI failure seems unrealated:

I just fixed the CI, tomorrow morning after 10am .. consider rebasing the topic.

File: Libs/MRML/Core/Testing/vtkMRMLSceneTest1.cxx [hunk 2]
```diff
@@ -38,7 +38,7 @@ class vtkMRMLCustomNode
   int ResetCount{0};
 
 protected:
-  vtkMRMLCustomNode(){}
+  vtkMRMLCustomNode()= default;
```
Comment (jcfr) [inline Libs/MRML/Core/Testing/vtkMRMLSceneTest1.cxx:41]: While we are updating the line, replace with `()= default;` -> `() = default;`

File: Modules/Loadable/VolumeRendering/MRMLDM/vtkMRMLVolumeRenderingDisplayableManager.cxx (hunk 1 above)
Comment (hjmjohnson) [reply Modules/Loadable/VolumeRendering/MRMLDM/vtkMRMLVolumeRenderingDisplayableManager.cxx:1082]: @jcfr. An autoformatter is really needed.  We should employ clang-format or similar if that level of formatting consistency is desired.

**Comment by jcfr** [comment]
I tested locally and there are no issues.

To update the code and add `= delete` and `= default`, did you use approaches documented here:
* [C++11: Update source code to use = delete](https://www.slicer.org/wiki/Documentation/Nightly/Developers/Tutorials/MigrationGuide#C.2B.2B11:_Update_source_code_to_use_.3D_delete)
* [C++11: Update source code to use = default](https://www.slicer.org/wiki/Documentation/Nightly/Developers/Tutorials/MigrationGuide/ObsoleteCodeRemoval#C.2B.2B11%3A_Update_source_code_to_use_.3D_default)

> STYLE: Use default member initialization

How did you implement the corresponding changes ?

**Comment by hjmjohnson** [comment]
Similar to above, but with manual interventions.

**Comment by hjmjohnson** [comment]
@jcfr can it be merged?

**Comment by jcfr** [comment]
> > >  STYLE: Use default member initialization

> > How did you implement the corresponding changes ?

> Similar to above, but with manual interventions.

Great. If you post here what are roughly the steps to do the update, I will add an entry on the wiki and then merge the  PR.

Documenting the process allow us to apply similar changes for extensions.

Thanks so much

**Comment by hjmjohnson** [comment]
git diff, look for spaces, remove spaces with vim.  I did not spend time writing a script,  I spend 15 minutes visually inspecting the spaces.

**Comment by hjmjohnson** [comment]
@jcfr any more concerns with this?  After it is merged, I'll start on the next batch of cleaning.

**Comment by jcfr** [comment]
To clarify, I was talking about this change. I am curious what is the process ? 

```
STYLE: Use default member initialization

Converts a default constructor’s member initializers into the new
default member initializers in C++11. Other member initializers that match the
default member initializer are removed. This can reduce repeated code or allow
use of ‘= default’.
```

**Comment by hjmjohnson** [comment]
> To clarify, I was talking about this change. I am curious what is the process ?
> 
> ```
> STYLE: Use default member initialization
> 
> Converts a default constructor’s member initializers into the new
> default member initializers in C++11. Other member initializers that match the
> default member initializer are removed. This can reduce repeated code or allow
> use of ‘= default’.
> ```

Using clang version 11.0.0 
https://clang.llvm.org/extra/clang-tidy/checks/modernize-use-default-member-init.html

**Comment by jcfr** [comment]
Thanks, migration guide entry added ... https://www.slicer.org/w/index.php?title=Documentation/Nightly/Developers/Tutorials/MigrationGuide#C.2B.2B11:_Use_default_member_initialization

**Comment by jcfr** [comment]
Will merge after local build completes, and after confirming wrapping of VTK based classes still work as expected.

(Indeed, considering that we still use an older VTK version where we backport changes, the wrapping infrastructure may be slightly out of date. The good news is that we are actively working on address this by supporting build against latest VTK version)

**Comment by jcfr** [comment]
Thanks for your patience @hjmjohnson :pray:

**Pull request merged**